### PR TITLE
[webview_flutter] Support js alert, confirm, prompt for android

### PR DESCRIPTION
**Pull request by dieudv**
## Description

Add support for javascript `alert`, `confirm`, `prompt` on android.

## Related Issues

* flutter/flutter#30358

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [ ] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [x] All existing and new tests are passing.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] The analyzer (`flutter analyze`) does not report any problems on my PR.
- [x] I read and followed the [Flutter Style Guide].
- [x] The title of the PR starts with the name of the plugin surrounded by square brackets, e.g. [shared_preferences]
- [x] I updated pubspec.yaml with an appropriate new version according to the [pub versioning philosophy].
- [x] I updated CHANGELOG.md to add a description of the change.
- [x] I signed the [CLA].
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (please indicate a breaking change in CHANGELOG.md and increment major revision).
- [x] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/plugins/blob/master/CONTRIBUTING.md
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[pub versioning philosophy]: https://www.dartlang.org/tools/pub/versioning
[CLA]: https://cla.developers.google.com/

### CODE REVIEW

File: packages/webview_flutter/android/src/main/java/io/flutter/plugins/webviewflutter/InputAwareWebView.java [hunk 1]
```diff
@@ -27,8 +27,8 @@
   private View threadedInputConnectionProxyView;
   private ThreadedInputConnectionProxyAdapterView proxyAdapterView;
 
-  InputAwareWebView(Context context, View containerView) {
-    super(context);
+  InputAwareWebView(Activity activity, View containerView) {
+    super(activity);
```
Comment (amirh) [inline packages/webview_flutter/android/src/main/java/io/flutter/plugins/webviewflutter/InputAwareWebView.java]: We must instantiate the WebView with the presentation context, the activity context isn't tuned for the VD size, doesn't use our custom window manager, and won't use our IMM wrapper.

**Comment by dieudv** [comment]
So any idea to fix #30358? I have been waiting for this issue for so long.

**Comment by amirh** [comment]
> So any idea to fix #30358? I have been waiting for this issue for so long.

I haven't looked at it, but my hunch would be that it may be implemented as Android popup windows which I believe are doing some short-circuiting based on window focus checks which we may be able to tweak.

**Comment by amirh** [comment]
see some more details in https://github.com/flutter/flutter/issues/24584#issuecomment-520162713 (which I suspect may be the same root cause, though I might be wrong)

**Pull request closed without merging**